doc/wcmp/wcmp-design.md

```diff
@@ -0,0 +1,605 @@
+factors the bandwidth value of a particular path against the total bandwidth values of all possible paths,  
+normalized to the range 1 to 255
+* FRR converts the incoming BGP link bandwidth extended community values into proportionated weight  
+among the ecmp members in such a way that the cumulative value of individual weights is normalized to 255
+* BGP gets notified about the weight change for the prefixes towards the switch that experienced the failure  
+of the link connecting the Spine and the remote ToR
+* FRR/Zebra recalculates the weight and makes a new NHG
+* NH weights on routes are reported to `fpmsyncd` and further via Redis DB to Route OA
+* Route OA programs weight values to `syncd` via SAI Redis
+* ASIC is provisioned with weight values by `syncd` via SAI API
+
+The NHG member weights can be used by a dataplane to do better forwarding decisions.
+
+## 2.2 FRR WCMP configuration
+
+The configuration for weighted ECMP using BGP link bandwidth requires using a route-map to inject  
+the link bandwidth extended community.
+
+There is no configuration necessary to process received link bandwidth and translate it into the weight  
+associated with the corresponding next hop; that happens by default. If some of the multipaths do not have  
+the link bandwidth extended community, the default behavior is to revert to normal ECMP.
+
+At the entry point router that is injecting the prefix to which weighted load balancing must be performed,  
+a route-map must be configured to attach the link bandwidth extended community.
+
+For the use case of providing weighted load balancing for an anycast service, this configuration will typically  
+need to be applied at the ToR or Leaf router that is connected to servers which provide the anycast service  
+and the bandwidth would be based on the number of multipaths for the destination.
+
+### 2.2.1 L3 network
+
+**Skeleton code:**
+```vtysh
+!
+route-map wcmp-map permit 100
+  set extcommunity bandwidth num-multipaths
+exit
+!
+router bgp 65100
+  neighbor SPINE peer-group
+  neighbor 1.1.1.1 peer-group SPINE
+  neighbor 1.1.1.1 remote-as 65200
+  neighbor 2.2.2.2 peer-group SPINE
+  neighbor 2.2.2.2 remote-as 65200
+  !
+  address-family ipv4 unicast
+    neighbor SPINE route-map wcmp-map out
+    neighbor SPINE activate
+  exit-address-family
+  !
+end
+```
+
+### 2.2.2 EVPN network
+
+**Skeleton code:**
+```vtysh
+!
+route-map wcmp-map permit 100
+  set extcommunity bandwidth num-multipaths
+exit
+!
+router bgp 65100 vrf vrf1
+  neighbor SPINE peer-group
+  neighbor 1.1.1.1 peer-group SPINE
+  neighbor 1.1.1.1 remote-as 65200
+  neighbor 2.2.2.2 peer-group SPINE
+  neighbor 2.2.2.2 remote-as 65200
+  !
+  address-family l2vpn evpn
+    advertise ipv4 unicast route-map wcmp-map
+    neighbor SPINE activate
+  exit-address-family
+  !
+end
+```
+
+**Note:** EVPN configuration is only relevant when `docker_routing_config_mode` is either `split` or `split-unified`
+
+## 2.3 Orchestration agent
```

Comment:
We need to rename this section as this section talks about programming the WCMP configs into the FRR via bgpcfgd.

You can link the existing the WCMP HLD created with orchestration agent changes here.

Reply:
@venkatmahalingam done